engine.go

```diff
@@ -60,9 +66,39 @@ func (eng *Engine) Handlers() []Handler {
 // To prevent any deadlock from happening this method should never be called
 // from the handler's HandleMetric method.
 func (eng *Engine) Register(handler Handler) {
-	eng.hmutex.Lock()
+	eng.mutex.Lock()
 	eng.handlers = append(eng.handlers, handler)
-	eng.hmutex.Unlock()
+	eng.mutex.Unlock()
+}
+
+// HistogramBuckets returns a map of metric names to buckets used to distribute
+// hitogram values.
+//
+// The buckets are of list of upper limits used to group the observed values.
+func (eng *Engine) HistogramBuckets() map[string][]float64 {
+	eng.mutex.RLock()
+	buckets := make(map[string][]float64, len(eng.buckets))
```

Comment:
If we do we can't set the size hint because accessing `len(eng.buckets)` requires holding the read lock.